polkadot/runtime/westend/src/lib.rs

```diff
@@ -2242,6 +2245,34 @@ sp_api::impl_runtime_apis! {
 		}
 	}
 
+	impl xcm_fee_payment_runtime_api::dry_run::XcmDryRunApi<Block, RuntimeCall, RuntimeEvent> for Runtime {
+		fn dry_run_extrinsic(extrinsic: <Block as BlockT>::Extrinsic) -> Result<ExtrinsicDryRunEffects<RuntimeEvent>, XcmDryRunApiError> {
+			use xcm_builder::InspectMessageQueues;
+			pallet_xcm::ShouldRecordXcm::<Runtime>::put(true);
+			let result = Executive::apply_extrinsic(extrinsic).map_err(|error| {
+				log::error!(
+					target: "xcm::XcmDryRunApi::dry_run_extrinsic",
+					"Applying extrinsic failed with error {:?}",
+					error,
+				);
+				XcmDryRunApiError::InvalidExtrinsic
+			})?;
+			let local_xcm = pallet_xcm::RecordedXcm::<Runtime>::get().unwrap_or(Xcm(Vec::new()));
+			let forwarded_messages = xcm_config::XcmRouter::get_messages();
```

Comment:
@mrshiposha Created a trait that can be used for easily implementing this API in live runtimes